Conflicting declaration of 'char* strndup' with 'C' linkage

**OS:** Alpine Linux (Edge)
**Version:** 0.2.1
**Commit/Build:** 8ac731e2124ecfb43f592c7f1cc5dd6902d5d83f

Probably important to note, this system uses musl rather than glibc. Trying to build OpenRCT2 for it fails at the very start:

```
[  0%] Building CXX object CMakeFiles/libopenrct2.dir/src/openrct2/Cheats.cpp.o
In file included from /usr/include/fortify/wchar.h:24,
                 from /usr/include/c++/8.2.0/cwchar:44,
                 from /usr/include/c++/8.2.0/bits/postypes.h:40,
                 from /usr/include/c++/8.2.0/iosfwd:40,
                 from /usr/include/c++/8.2.0/memory:72,
                 from /home/pmos/build/src/OpenRCT2-0.2.1/src/openrct2/GameState.h:14,
                 from /home/pmos/build/src/OpenRCT2-0.2.1/src/openrct2/Cheats.cpp:12:
/usr/include/string.h:69:7: error: conflicting declaration of 'char* strndup(const char*, size_t)' with 'C' linkage
 char *strndup (const char *, size_t);
       ^~~~~~~
In file included from /home/pmos/build/src/OpenRCT2-0.2.1/src/openrct2/Cheats.h:13,
                 from /home/pmos/build/src/OpenRCT2-0.2.1/src/openrct2/Cheats.cpp:10:
/home/pmos/build/src/OpenRCT2-0.2.1/src/openrct2/common.h:85:7: note: previous declaration with 'C++' linkage
 char* strndup(const char* src, size_t size);
       ^~~~~~~
make[2]: *** [CMakeFiles/libopenrct2.dir/build.make:63: CMakeFiles/libopenrct2.dir/src/openrct2/Cheats.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/libopenrct2.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```


- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Install Alpine Linux (for example in a VM)
2. Install the required dependencies: `apk add cmake jansson-dev libzip-dev libpng-dev curl-dev fontconfig-dev icu-dev sdl2-dev speexdsp-dev mesa-dev`
3. Try to build the program

--- a/src/openrct2/common.h
+++ b/src/openrct2/common.h
@@ -79,11 +79,9 @@ using colour_t = uint8_t;
 
 #endif // defined(__unix__) || (defined(__APPLE__) && defined(__MACH__))
 
-#if !(                                                                                                                         \
-    (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200809L) || (defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 700)               \
-    || (defined(__APPLE__) && defined(__MACH__)) || defined(__ANDROID_API__) || defined(__FreeBSD__))
+#ifdef _WIN32
 char* strndup(const char* src, size_t size);
-#endif // !(POSIX_C_SOURCE >= 200809L || _XOPEN_SOURCE >= 700)
+#endif
 
 // BSD and macOS have MAP_ANON instead of MAP_ANONYMOUS
 #ifndef MAP_ANONYMOUS

--- a/src/openrct2/platform/Shared.cpp
+++ b/src/openrct2/platform/Shared.cpp
@@ -43,9 +43,7 @@
 static mach_timebase_info_data_t _mach_base_info = {};
 #endif
 
-#if !(                                                                                                                         \
-    (defined(_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200809L) || (defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 700)               \
-    || (defined(__APPLE__) && defined(__MACH__)))
+#ifdef _WIN32
 char* strndup(const char* src, size_t size)
 {
     size_t len = strnlen(src, size);
@@ -60,13 +58,10 @@ char* strndup(const char* src, size_t size)
     dst[len] = '\0';
     return dst;
 }
-#endif // !((defined (_POSIX_C_SOURCE) && _POSIX_C_SOURCE >= 200809L) || (defined(_XOPEN_SOURCE) && _XOPEN_SOURCE >= 700) ||
-       // (defined(__APPLE__) && defined(__MACH__)))
 
-#ifdef _WIN32
 static uint32_t _frequency = 0;
 static LARGE_INTEGER _entryTimestamp;
-#endif
+#endif // _WIN32
 
 using update_palette_func = void (*)(const uint8_t*, int32_t, int32_t);
 

